core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestionBuilder.java

```diff
@@ -125,6 +122,26 @@ private CompletionSuggestionBuilder(String fieldname, CompletionSuggestionBuilde
     }
 
     /**
+     * Read from a stream.
+     */
+    public CompletionSuggestionBuilder(StreamInput in) throws IOException {
+        super(in);
+        payloadFields = new ArrayList<>();
```

Comment:
shall we make payloadFields an array instead? It seems possible given that we only set it.